src/backend/Backend.jsx

```diff
@@ -0,0 +1,102 @@
+// src/components/SomeComponent.js
+import React, { useState, useEffect } from "react";
+import {
+  createData,
+  readData,
+  updateData,
+  deleteData,
+} from "../utils/Functions/functions";
+
+const BackEndTest = () => {
+  const [data, setData] = useState([]);
+  const [error, setError] = useState(null);
+  const collectionName = "Users";
+
+  // fetch data and update state when component mounts
+  const fetchDataAndUpdateState = async () => {
+    setError(null);
+    try {
+      const result = await readData(collectionName);
+      setData(result);
+      console.log(result);
+    } catch (error) {
+      setError("Error fetching data");
+      console.error(error);
+    }
+  };
+
+  useEffect(() => {
+    fetchDataAndUpdateState();
+  }, []);
+
+  const handleCreate = async () => {
+    setError(null);
+    try {
+      const newData = { name: "New Item", description: "Description" };
+      await createData(collectionName, newData);
+      fetchDataAndUpdateState();
```

Comment:
So in create data you already have a try catch there. If the createDoc succeeds you don't need to refetch the data again because firebase would have ensured it's successful.

That means you don't need the try catch block here. instead you can just set state directly and avoid another refetch of the data.